test/e2e/framework.go

```diff
@@ -203,43 +204,42 @@ type PodIPs struct {
 	ipStrings []string
 }
 
-type deployAntreaOptions int
-
-const (
-	deployAntreaDefault deployAntreaOptions = iota
-	deployAntreaIPsec
-	deployAntreaWireGuardGo
-	deployAntreaCoverageOffset
-)
-
-func (o deployAntreaOptions) WithCoverage() deployAntreaOptions {
-	return o + deployAntreaCoverageOffset
+type deployAntreaOption struct {
+	name string
+	ymls []string
 }
 
-func (o deployAntreaOptions) DeployYML() string {
-	return deployAntreaOptionsYML[o]
+func (o *deployAntreaOption) withCoverage() {
+	var covYMLMap = map[string]string{
+		antreaYML:            antreaCovYML,
+		antreaIPSecYML:       antreaIPSecCovYML,
+		antreaWireGuardGoYML: antreaWireGuardGoCovYML,
+	}
+	for i := range o.ymls {
+		if _, ok := covYMLMap[o.ymls[i]]; ok {
+			o.ymls[i] = covYMLMap[o.ymls[i]]
+		}
+	}
 }
 
-func (o deployAntreaOptions) String() string {
-	return deployAntreaOptionsString[o]
+func (o deployAntreaOption) getYML() []string {
+	return o.ymls
 }
 
-var (
-	deployAntreaOptionsString = [...]string{
-		"AntreaDefault",
-		"AntreaWithIPSec",
-		"AntreaWithWireGuardGo",
-	}
-	deployAntreaOptionsYML = [...]string{
-		antreaYML,
-		antreaIPSecYML,
-		antreaWireGuardGoYML,
-		antreaCovYML,
-		antreaIPSecCovYML,
-		antreaWireGuardGoCovYML,
-	}
+const (
+	deployAntreaDefault int = iota
+	deployAntreaWindows
+	deployAntreaIPsec
+	deployAntreaWireGuardGo
 )
 
+var deployAntreaOptions = map[int]deployAntreaOption{
+	deployAntreaDefault:     {"AntreaDefault", []string{antreaYML}},
+	deployAntreaWindows:     {"AntreaWindows", []string{antreaYML, antreaWindowsYML}},
+	deployAntreaIPsec:       {"AntreaWithIPSec", []string{antreaIPSecYML}},
+	deployAntreaWireGuardGo: {"AntreaWithWireGuardGo", []string{antreaWireGuardGoYML}},
+}
```

Comment:
I don't think the integer values (`deployAntreaDefault `, `deployAntreaWindows`, ...) are needed. You could use the struct directly and remove the `deployAntreaOptions` map:
```go
var deployAntreaDefault = NewConfiguration("AntreaDefault", antreaYAML)
var deployAntreaWindows = NewConfiguration("AntreaWindows", antreaYML, antreaWindowsYML)
...
```

Reply:
Updated.